Sources/MapboxMaps/Foundation/Camera/GestureDecelerationCameraAnimator.swift

```diff
@@ -0,0 +1,68 @@
+import UIKit
+
+internal final class GestureDecelerationCameraAnimator: NSObject, CameraAnimatorInterface {
+
+    private var location: CGPoint
+    private var velocity: CGPoint
+    private let decelerationRate: CGFloat
+    private let locationChangeHandler: (CGPoint) -> Void
+    private var previousDate: Date?
+    private let dateProvider: DateProvider
+    internal var completion: (() -> Void)?
+
+    internal init(location: CGPoint,
+                  velocity: CGPoint,
+                  decelerationRate: CGFloat,
+                  locationChangeHandler: @escaping (CGPoint) -> Void,
+                  dateProvider: DateProvider) {
```

Comment:
We can provide `DefaultDateProvider` as a default argument to `dateProvider` here. It'll clean up the call site.

Reply:
Oh wait -- I see that the constructor for `PanGestureHandler` also takes a `dateProvider`. Is that necessary?

Reply:
Let's discuss this. I'd like to move away from dependency injection via default args, so while this does make the call site a bit messier, I see it as an intermediate state on the way towards a more clear composition root.

Reply:
Mhm okay!